src/portable/st/stm32f4/dcd_stm32f4.c

```diff
@@ -80,20 +80,39 @@ static void bus_reset(void) {
   dev->DOEPMSK |= USB_OTG_DOEPMSK_STUPM | USB_OTG_DOEPMSK_XFRCM;
   dev->DIEPMSK |= USB_OTG_DIEPMSK_TOM | USB_OTG_DIEPMSK_XFRCM;
 
+  // Peripheral FIFO architecture (Rev18 RM 29.11)
+  //
+  // --------------- 312.5 ( 1250 bytes )
+  // | IN FIFO 3  |
+  // --------------- y + x + 16 + GRXFSIZ
+  // | IN FIFO 2  |
+  // --------------- x + 16 + GRXFSIZ
+  // | IN FIFO 1  |
+  // --------------- 16 + GRXFSIZ
+  // | IN FIFO 0  |
+  // --------------- GRXFSIZ
+  // | OUT FIFO   |
+  // | ( Shared ) |
+  // --------------- 0
+  //
   // FIFO sizes are set up by the following rules (each word 32-bits):
-  // OUT FIFO uses (based on page 1354 of Rev 17 of reference manual):
+  // All EP OUT shared a unique OUT FIFO which uses (based on page 1354 of Rev 17 of reference manual):
   // * 10 locations in hardware for setup packets + setup control words
   // (up to 3 setup packets).
   // * 2 locations for OUT endpoint control words.
-  // * 64 bytes for maximum control packet size.
+  // * 16 + 1 (data + info) for largest packet size of 64 bytes.
   // * 1 location for global NAK (not required/used here).
-  // IN FIFO uses 64 bytes for maximum control packet size.
   //
   // However, for OUT FIFO, 10 + 2 + 16 = 28 doesn't seem to work (TODO: why?).
   // Minimum that works in practice is 35, so allocate 40 32-bit locations
   // as a buffer.
-  USB_OTG_FS->GRXFSIZ = 40;
-  USB_OTG_FS->DIEPTXF0_HNPTXFSIZ |= (16 << USB_OTG_TX0FD_Pos); // 16 32-bit words = 64 bytes
+  //
+  // It is recommended to allocate 2 times the largest packet size, therefore
+  // Recommended value = 10 + 1 + 2 x (16+1) = 45 --> Let's make it 50
```

Comment:
Because each OUT packet will have two control words, I think this comment should _more accurately_ be the following: `10 + 1 + 2 x (16+2) = 47`. Doesn't change the `GRXFSIZ` you chose; I just want the comment to be consistent :).